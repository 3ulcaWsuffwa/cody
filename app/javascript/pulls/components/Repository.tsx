import React from "react";
import { Link } from "react-router-dom";
import { createFragmentContainer, graphql } from "react-relay";
import type { Repository_repository } from "./__generated__/Repository_repository.graphql";
import { ChevronRight, Folder } from "react-feather";

function Repository({
  repository,
}: {
  repository: Repository_repository;
}): JSX.Element {
  return (
    <li>
      <Link
        to={`/repos/${repository.owner}/${repository.name}`}
        className="block hover:bg-gray-50"
      >
        <div className="flex items-center px-4 py-4 sm:px-6">
          <div className="min-w-0 flex-1 flex items-center">
            <div className="flex-shrink-0">
              <Folder />
            </div>
            <div className="min-w-0 flex-1 px-4 md:grid md:grid-cols-2 md:gap-4">
              <div>
                <p className="text-sm font-medium truncate">{`${repository.owner}/${repository.name}`}</p>
              </div>
            </div>
            <div>
              <ChevronRight />
            </div>
          </div>
        </div>
      </Link>
    </li>
    // <div className="level box">
    //   <div className="level-left code">
    //     <div className="level-item">
    //       <strong>{`${repository.owner}/${repository.name}`}</strong>
    //     </div>
    //   </div>
    //   <div className="level-right">
    //     <div className="level-item">
    //       <Link
    //         to={`/repos/${repository.owner}/${repository.name}`}
    //         className="button"
    //         title={`${repository.owner}/${repository.name}`}
    //       >
    //         <MoreHorizontal />
    //       </Link>
    //     </div>
    //   </div>
    // </div>
  );
}

export default createFragmentContainer(Repository, {
  repository: graphql`
    fragment Repository_repository on Repository {
      id
      owner
      name
    }
  `,
});
