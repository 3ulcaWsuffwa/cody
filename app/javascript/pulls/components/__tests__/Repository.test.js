import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import Repository from "../Repository";

test("Repository snapshot test", () => {
  const props = {
    owner: "aergonaut",
    name: "testrepo"
  };
  const component = renderer.create(
    <MemoryRouter>
      <Repository repository={props} />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
