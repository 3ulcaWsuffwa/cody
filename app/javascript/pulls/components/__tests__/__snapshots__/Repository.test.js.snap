// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository snapshot test 1`] = `
<div
  className="level box"
>
  <div
    className="level-left code"
  >
    <div
      className="level-item"
    >
      <strong>
        aergonaut/testrepo
      </strong>
    </div>
  </div>
  <div
    className="level-right"
  >
    <div
      className="level-item"
    >
      <a
        className="button"
        href="/repos/aergonaut/testrepo"
        onClick={[Function]}
      >
        • • •
      </a>
    </div>
  </div>
</div>
`;
